### ISSUE TITLE
Request: Ring Alarm plugin

### ISSUE THREAD

**Issue by fitzage** [nfarina/homebridge#1931]
I know it's early days, but I want to at least a get this out there. The new Ring Alarm system is out, and I went ahead and bought one. They've told me they'll eventually have HomeKit support, but who knows when that will be.

I'm available if someone who actually knows how to build the plugin needs my help for testing or something.

**Comment by AppleTechy** [comment]
I imagine this would be bundled into a Ring Homebridge plugin. Since there currently isn’t even a plugin for the Doorbell or the Stickup cam/Floodlight

**Comment by fitzage** [comment]
There's actually a doorbell plugin, but it doesn't support the camera bits, although they link to another plugin that supposedly can do that.

I haven't felt a need for it, so I haven't tried it.

https://github.com/homespun/homebridge-platform-ring-video-doorbell

**Comment by AppleTechy** [comment]
If its the plugin I am thinking of it only pulls the video snapshots rather than allowing an actual live video stream.

**Comment by fitzage** [comment]
It looked like it only does like motion sensing to trigger other things or something.

**Comment by AppleTechy** [comment]
I started talking again to @Homespun the maintainer of that about adding the actual streams. The alarm system will be difficults since Ring doesn't offer a Public API and neither of have it. So we have been looking at https://github.com/jimhigson/ring-api to see if he adds anything relating to the alarm

**Comment by cah-dgreif** [comment]
I've published [`homebridge-ring-alarm`](https://github.com/dgreif/homebridge-ring-alarm) using my [fork of ring-api](https://github.com/dgreif/ring-api) for now.  It's my first attempt at a homebridge plugin so any feedback/issues/PRs are appreciated!

**Comment by fitzage** [comment]
@cah-dgreif It's working great for me so far! I wasn't expecting the direct access to specific sensors, but I'll have to play with that and see if it might be useful to me.

**Comment by dgreif** [comment]
@fitzage Yeah, there's a ton of possibilities since you can have the sensors trigger other automation.  Like turning your lights on when you detect motion in the middle of the night.  The main feature that I was going for was automatically arming/disarming when I leave/arrive home.  All the other sensor stuff was just an easy add-on 😄

**Comment by jamesg311** [comment]
@dgreif can you explain what the different Arm states in HomeKit align with for your Ring Alarm plug-in?

I successfully added the plugin last night and it detected all my sensors—which even work with motion detection for turning on lights via an automation I set up. 

However, i set my ring Alarm to “Home” via Ring App and this status wasn’t reflected in the Home app. Maybe there is a lag...I didn’t wait long before disarming the system. But I’m hesitant to set the Alarm to a mode from the Home app if it might trigger my Alarm system to blast the siren. 

Thanks.

**Comment by dgreif** [comment]
@jamesg311 I've added a [section to the Readme](https://github.com/dgreif/homebridge-ring-alarm#alarm-modes) to clarify how the alarm modes work.  Let me know if you have any more questions, or feel free to open an issue if something seems to not be working.

**Comment by weihanwu** [comment]
New to this... hope this question makes sense

I have homebridge on rpi running and connected to HomeKit.  Then added homebridge-smartthings-tonesto plugin - worked great.  Then added homebridge-nest plugin - worked great.  Then tried adding homebridge-ring-alarm - not working.  Am I suppose to first somehow install ring-api on rpi?

Thanks for any guidance!

**Comment by dgreif** [comment]
@weihanwu Sorry to hear you are having issues getting the plugin installed.  It does require node 8.3.0 or higher, which may be what's causing issues for you.  Run `node -v` to see which version you currently have installed.  There is no need to install `ring-api` separately, just `npm i -g homebridge-ring-alarm`.  If you continue to have issues, please open an issue on the [homebridge-ring-alarm](https://github.com/dgreif/homebridge-ring-alarm) repo.  @nfarina you can probably close this issue now that the plugin is fully operational.

**Comment by gadgetfreak98** [comment]
I'm getting this error in trying to install the plugin:  

pi@raspberrypi:~ $ sudo npm i -g homebridge-ring-alarm
npm ERR! code 128
npm ERR! Command failed: /usr/bin/git submodule update -q --init --recursive
npm ERR! fatal: Could not change back to '/root/.npm/_cacache/tmp/git-clone-884d9ccd': Permission denied

Any ideas?  I do have other successfully running plugins.  

One thing that might be the issue is that I have Node 8.15.  I did update to a later version (10.something) and tried to update then, but got the same error.  Also, my homebridge setup stopped working with that update, so I went back to 8.15.  But again, it didn't allow for installing the ring-alarm plugin even with the higher version.  Thoughts?
